hibernate-core/src/main/java/org/hibernate/query/results/complete/CompleteResultBuilderBasicValuedStandard.java

```diff
@@ -166,14 +166,14 @@ public boolean equals(Object o) {
 		if ( !Objects.equals( explicitType, that.explicitType ) ) {
 			return false;
 		}
-		return explicitJavaType.equals( that.explicitJavaType );
+		return Objects.equals( explicitJavaType, that.explicitJavaType );
 	}
 
 	@Override
 	public int hashCode() {
 		int result = explicitColumnName != null ? explicitColumnName.hashCode() : 0;
 		result = 31 * result + ( explicitType != null ? explicitType.hashCode() : 0 );
-		result = 31 * result + explicitJavaType.hashCode();
+		result = 31 * result + ( explicitJavaType != null ? explicitJavaType.hashCode() : 0 );
 		return result;
```

Comment:
My initial change is to use Objects util methods aggressively as following:
```
		return Objects.equals( explicitColumnName, that.explicitColumnName )
				&& Objects.equals( explicitType, that.explicitType )
				&& Objects.equals( explicitJavaType, that.explicitJavaType );
```
and
```
return Objects.hash( explicitColumnName, explicitType, explicitJavaType );
```

but later on I reverted back the above changes for I think maybe you have some special reason to avoid the approach (consistent and error-free to me) for performance reason?